### ISSUE TITLE
Create a user flow for outgoing donations

### ISSUE THREAD

**Issue by HaGuesto** [boxwise/boxwise_wms#30]
## Context
Describe the user flow how boxes of donations leave a warehouse.

## ToDos
The following steps should be included:
- coordinator orders boxes from outside the warehouse.
- pick lists are visible on the mobile screen for volunteers.
- boxes are picked.
- boxes cannot be found.
- boxes are actually moved.

Please use horizontal swimlanes for the different user groups!

**Comment by jamescrowley** [comment]
@HaGuesto Looking at the diagram you included, we have another persona - the customer / external organisation. I'll update our personas in https://github.com/boxwise/boxwise_wms/issues/29 - but I suspect they are 'out of scope' for our initial milestone, and we just leave it as a responsibility of the co-ordinator for now to do any external communication?

**Comment by HaGuesto** [comment]
@jamescrowley Exactly right. For the moment the only person who can put an order together should be the coordinator.

**Comment by jamescrowley** [comment]
@HaGuesto the 'interrupt' points in the existing diagrams are interesting (when a box can't be found, and alternates are suggested). 

I wonder if ideally the co-ordinator would not be ordering specific boxes (as they do in the current app), but ‘I need 20 of these product variants’ and the system matches it as best they can? This might be overly complex right now but just wanted to plant the thought.

Either way, the way I'm thinking about the 'lost/alternate' aspect is as a 'substitution' in the way supermarket home deliveries work - flag it when the 'pick' is complete and the co-ordinator can then decide to reject the replacement or accept it. 

I think that follows the 'real-life' workflow anyway - the volunteer on the ground is going to make a call, and then the coordinator only gets to rectify much later.

Make sense?

**Comment by jamescrowley** [comment]
On lost boxes - when at the camp, volunteers were reluctant to mark a box as lost until someone else had also looked for it. I think I'd mark it as 'possibly lost' - and then have a separate process for users to review possibly lost boxes. Or, integrate it into the picking process itself, but that would require a minimum of two volunteers in the warehouse. Thoughts?

**Comment by jamescrowley** [comment]
First draft:

![outgoing donations](https://user-images.githubusercontent.com/509533/52795341-dedd6600-3071-11e9-9d81-3ca7d0eecb7b.png)

**Comment by HaGuesto** [comment]
> @HaGuesto the 'interrupt' points in the existing diagrams are interesting (when a box can't be found, and alternates are suggested).
> 
> I wonder if ideally the co-ordinator would not be ordering specific boxes (as they do in the current app), but ‘I need 20 of these product variants’ and the system matches it as best they can? This might be overly complex right now but just wanted to plant the thought.
> 
> Either way, the way I'm thinking about the 'lost/alternate' aspect is as a 'substitution' in the way supermarket home deliveries work - flag it when the 'pick' is complete and the co-ordinator can then decide to reject the replacement or accept it.
> 
> I think that follows the 'real-life' workflow anyway - the volunteer on the ground is going to make a call, and then the coordinator only gets to rectify much later.
> 
> Make sense?

I have never done a supermarket home delivery. So, I do not get it completely.
I would never let the system decide which box to take as a replacement. The best we can do is offer him a list of other boxes with the same product at the same location and the volunteer can choose which one to pick.

**Comment by HaGuesto** [comment]
> On lost boxes - when at the camp, volunteers were reluctant to mark a box as lost until someone else had also looked for it. I think I'd mark it as 'possibly lost' - and then have a separate process for users to review possibly lost boxes. Or, integrate it into the picking process itself, but that would require a minimum of two volunteers in the warehouse. Thoughts?

I agree there should be a staging area for lost boxes and a separate workflow.

**Comment by HaGuesto** [comment]
## Comments about the process flow

- 1st step: Could you be a bit more precise how a coordinator puts together an order? Does he/she do it on the phone or laptop? 
I'm asking this because we need to figure out which data we should offer them for the selection. Might the age of the box or a condition tag (new or used) interesting for them?
- 2nd step: Does this mean that only one person can work on an order?
- step "Physically put the boxes in a staging area". The question you are asking is a organizational problem which should be solved by the user. I do not think we can control that through the app.
- App action "Box marked as possibly lost": 
     a) I do not get the second sentence.
     b) As mentioned above Boxwise should show a list of alternatives. 
- App action "marked as Substituted. ah now I get it. :) The substitution tracking. quite advanced. mmh we have to check if that is really needed.
- App action "Order marked as fully picked". What triggers this action? Does this mean that the volunteer can always just see one box to pick at a time? And when the last one is picked or possibly lost it does just triggers it?
- the last three steps: I do not want to overcomplicate this process --> for a box there should be just two stages (ordered and picked), for a picking only three (new, in progress, finished). If the coordinator checks the picking or not should be up to them --> again something organizational.
Besides that we have to figure out how we get all boxes out of the "picked" stage and actually register them in a new location / as given out. At the moment each box needs to be scanned again and be moved to resolve the "picked" stage. Do you think we should offer one button (but just for the coordinator) which moves out all boxes at ones? Should we still offer the old workflow?

**Comment by jamescrowley** [comment]
> ## Comments about the process flow
> * 1st step: Could you be a bit more precise how a coordinator puts together an order? Does he/she do it on the phone or laptop?
>   I'm asking this because we need to figure out which data we should offer them for the selection. Might the age of the box or a condition tag (new or used) interesting for them?
I saw both being done.
> * 2nd step: Does this mean that only one person can work on an order?
I think we should probably allow multiple people. But then you may want to record when a volunteer has chosen a box to go pick (in progress), so you only show the unpicked boxes.
> * step "Physically put the boxes in a staging area". The question you are asking is a organizational problem which should be solved by the user. I do not think we can control that through the app.
Agreed

> * App action "Box marked as possibly lost":
>   a) I do not get the second sentence.
Unfulfilled - ie we weren't able to supply this box. I'm kind of 'defaulting' to this, and switching to substituted if we find a suitable replacement.
>   b) As mentioned above Boxwise should show a list of alternatives.
isn't this covered by 'Choose an alternative box offered by the app' ? Possibly there should be a 'Don't want any alternatives' route there I guess.
> * App action "marked as Substituted. ah now I get it. :) The substitution tracking. quite advanced. mmh we have to check if that is really needed.
> * App action "Order marked as fully picked". What triggers this action? Does this mean that the volunteer can always just see one box to pick at a time? And when the last one is picked or possibly lost it does just triggers it?
The volunteer could still see all the boxes - moving from the 'to pick' list to the 'picked' list, but yes I think once the 'to pick' list is empty, you automatically mark it as fully picked/finished?
> * the last three steps: I do not want to overcomplicate this process --> for a box there should be just two stages (ordered and picked), for a picking only three (new, in progress, finished). If the coordinator checks the picking or not should be up to them --> again something organizational.
👍 
>   Besides that we have to figure out how we get all boxes out of the "picked" stage and actually register them in a new location / as given out. At the moment each box needs to be scanned again and be moved to resolve the "picked" stage. Do you think we should offer one button (but just for the coordinator) which moves out all boxes at ones? Should we still offer the old workflow?

I was kinda thinking of the whole order as in transit / transferred to the new location (rather than individual items) - but you need to track the substitutions for that to work. But otherwise the process (at least in my experience) was quite a pain when having to re-scan every individual box and mark it as moved.

**Comment by jamescrowley** [comment]
![outgoing donations](https://user-images.githubusercontent.com/509533/52874101-eaec2500-3150-11e9-8649-353ac6fb291a.png)
